src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -27,18 +29,19 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
         const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${props.policy?.id ?? '0'}`, {
             initWithStoredValues: false,
         });
+        const isConnectionDataFetchNeeded = !isOffline && props.policy && props.policy.areConnectionsEnabled && hasConnectionsDataBeenFetched && props.policy.connections;
 
         useEffect(() => {
             // When the accounting feature is not enabled, or if the connections data already exists,
             // there is no need to fetch the connections data.
-            if (!props.policy || !props.policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!props.policy.connections) {
+            if (!isConnectionDataFetchNeeded || !props?.policy?.id) {
                 return;
             }
 
             openPolicyAccountingPage(props.policy.id);
-        }, [hasConnectionsDataBeenFetched, props.policy, isOffline]);
+        }, [hasConnectionsDataBeenFetched, props.policy, isOffline, isConnectionDataFetchNeeded]);
 
-        if (props.policy?.areConnectionsEnabled && (!props.policy || status === 'loading' || hasConnectionsDataBeenFetched === false)) {
+        if (props.policy?.areConnectionsEnabled && status === 'loading') {
```

Comment:
I'm still trying to see in which cases we will end up using the loader. The status initially will be loaded because initWithStoredValues is false

Reply:
seems like that we do not have any choice to show skeleton instead of loader. Just if we put back hasConnectionsDataBeenFetched === false - page with Accounting will not be loaded

Reply:
probably we can close this one @s77rt and close related GH issue as well

Reply:
The problem is that on other pages we are not using any loader indicator

Reply:
You can add a second parameter to withPolicyConnections to decide if we want to use the full screen loader

```js
function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(WrappedComponent: ComponentType<TProps>, shouldBlockView = true) {
```

Use that parameter with the rendering condition and in PolicyAccountingPage pass that parameter as false

Reply:
Yeah that works! @s77rt